### PR TITLE
fixes for freebsd cron invocation and kitchen changes required to test

### PR DESCRIPTION
**Pull request by lamont**
the cron recipe would not converge under freebsd, there's no init.d equiv and I thought the cleanest way would be to put a pair of not_if guards on the two unneeded templates. I also had to add the '"/bin/sh"' to the .kitchen.yml to get kitchen to work at all with the freebsd box files. I tested that the freebsd/cron suite works under vagrant and that my '"/bin/sh"' changes do not break at least the ubuntu suites.

### CODE REVIEW

**Comment by lamont** [comment]
I note that travis is and has been failing but PR #337 fixes that issue. My local kitchen tests succeeded, but it would be nice to get the green travis badge of happiness as well.

**Comment by lamont-granquist** [comment]
lol, confusing...

**Comment by lamont** [comment]
let's combine our lamont powers for great justice!  But on a more serious note, I'm happy to yank out my .travis.yml changes if the other PR does a better job. the .kitchen.yml change was required to get freebsd working under kitchen.

File: recipes/cron.rb [hunk 1]
```diff
@@ -46,11 +46,13 @@ class ::Chef::Recipe
     source "#{dist_dir}/init.d/chef-client.erb"
     mode 0755
     variables(client_bin: client_bin)
+    not_if { node['platform_family'] == 'freebsd'}
```
Comment (lamont-granquist) [inline recipes/cron.rb]: that should probably not be a not_if and should wrap the template completely so it doesn't even appear as a resource on freebsd.

foodcritic might get annoyed by this, but foodcritic can kinda suck it and you can use `# ~FCXYZ` to turn it off.

File: .kitchen.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@ driver:
   customize:
     memory: 1024
     cpus: 2
+  ssh:
+    shell: '"/bin/sh"'
```
Comment (lamont-granquist) [inline .kitchen.yml]: doesn't TK take driver arguments under the platform section as well, so that we can set this only for freebsd?

**Comment by lamont-granquist** [comment]
I don't have a lot of opinion on `rake spec` vs `rake travis`, it should probably `bundle exec` either way for less confusion.

**Comment by lamont** [comment]
Ok, I'll rework the .kitchen.yml change so that the ssh/shell is applied to the freebsd platform specifically and put the two templates in an anti-freebsd block then ping you again. Probably Thursday or Friday.  And I'll add the .foodcritic ignore for the warning I'm introducing.

**Comment by lamont** [comment]
Well, that fixes rake spec on my machine, but the TravisCI build fails cause we hit the 4 megabyte log limit due to the repeated warnings about needing override: true in new resources. 

WARN: You declared a new resource #<Class:0x007fbd95c3d430> for resource windows_certificate, but it comes alphabetically after LWRP resource windows_certificate from cookbook windows and has the same filters ({}), so it will not be used. Use override: true if you want to use it for windows_certificate.

I have no idea why that warning suddenly appeared, but it's affected other tests I was running using the apt cookbook. Happened under chefDK 0.7.0 and also 0.8.1.

**Comment by josephholsten** [comment]
@lamont did this ever get complete to your satisfaction?

**Comment by lamont-granquist** [comment]
looks fine to me but looks like it needs a serious rebase.

**Comment by dheerajd-msys** [comment]
@lamont Please check ^

**Comment by tas50** [comment]
I'm going to close this one out at this point due to the age and drift in the codebase. If this is still something you want to do let me know in the community slack and we can figure out next steps.

**Pull request closed without merging**